js/ui/shellDBus.js

```diff
@@ -592,24 +604,23 @@ const AppLauncher = new Lang.Class({
     },
 
     LaunchAsync: function(params, invocation) {
-        if (name == 'eos-app-store') {
+        let [appName, timestamp] = params;
+
+        if (appName == 'eos-app-store') {
             Main.appStore.show(timestamp, true);
+            Main.appStore.appLaunched = true;
```

Comment:
My feeling here is that the old behaviour is wrong - there's no reason to touch Main.appStore.appLaunched if we're launching some other app. That said, I'll separate it out.